tour_model_eval/user_input_request_times_all_users.ipynb

```diff
@@ -0,0 +1,568 @@
+    "#                     bin.append(trip_index)\n",
+    "#                     added = True\n",
+    "#                     break\n",
+    "#             if month:\n",
+    "#                 if evaluation.match_month(trip,bin,filter_trips):\n",
+    "#                     bin.append(trip_index)\n",
+    "#                     added = True\n",
+    "#                     break\n",
+    "\n",
+    "#         if not added:\n",
+    "#             bin_date.append([trip_index])\n",
+    "\n",
+    "#     return bin_date       "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "formed-drive",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "# build a base dataframe for request times in January \n",
+    "date_df = pd.DataFrame(data = {'date':np.arange(1,32),'drop_col':np.arange(1,32)})\n",
+    "date_df.set_index(['date'], inplace=True)\n",
+    "\n",
+    "# get valid user list\n",
+    "user_ls,valid_users = evaluation.get_user_ls(all_users,radius)\n",
+    "\n",
+    "for a in range(len(all_users)):\n",
+    "    user = all_users[a]\n",
+    "    filter_trips,sim,trips = evaluation.filter_data(user,radius)\n",
+    "    logging.debug(\"len(filter_trips)is %s \"% len(filter_trips))\n",
+    "\n",
+    "    # filter out users that don't have enough valid labeled trips\n",
+    "    if not evaluation.valid_user(filter_trips,trips):\n",
+    "        req_day.append(NaN)\n",
+    "        req_month.append(NaN) \n",
+    "        continue\n",
+    "    sim.bin_data()\n",
+    "    sim.delete_bins()\n",
+    "    bins = sim.bins\n",
+    "    \n",
+    "    # collect requested trips and common trips(no need to request) indices above cutoff\n",
+    "    ab_trip_ls = []\n",
+    "    no_req_trip_ls = []\n",
+    "    for bin in bins:\n",
+    "        early_trip = filter_trips[bin[0]]\n",
+    "        index = 0\n",
+    "        for i in range(1,len(bin)):\n",
+    "            compare_trip = filter_trips[bin[i]]\n",
+    "            if early_trip['data']['start_local_dt']['year']>compare_trip['data']['start_local_dt']['year']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "            elif early_trip['data']['start_local_dt']['year']==compare_trip['data']['start_local_dt']['year'] and early_trip['data']['start_local_dt']['month']>compare_trip['data']['start_local_dt']['month']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "            elif early_trip['data']['start_local_dt']['year']==compare_trip['data']['start_local_dt']['year'] and early_trip['data']['start_local_dt']['month']==compare_trip['data']['start_local_dt']['month'] and early_trip['data']['start_local_dt']['day']>compare_trip['data']['start_local_dt']['day']:\n",
+    "                early_trip = compare_trip\n",
+    "                index = i\n",
+    "        early_trip_index = bin[index]\n",
```

Comment:
Move this into `find_first_trip`?
Also, instead of this complicated logic to compare two trips based on the `start_local_dt`, you can compare directly based on the `start_ts`. You only need `start_local_dt` to group the trips and find the number of requests per day.

Reply:
I don't understand. Before this step, I group the trips based on clusters, this step is to look for the first trip. But why do I need to find the number of requests per day in one cluster?

Reply:
I updated this part in [644c2dd](https://github.com/e-mission/e-mission-eval-private-data/pull/19/commits/644c2dd1e64734d194270ccba0920f3bfe64d209)

Reply:
I don't understand. I have a cluster with trips t1, t2, t3. I want to find the first one. Why is this not just comparing `t1.start_ts`, `t2.start_ts` and `t3.start_ts`? Why do you need complicated logic involving the year, day and month?

Reply:
ah looks like you are using `start_ts` in https://github.com/e-mission/e-mission-eval-private-data/pull/19/commits/644c2dd1e64734d194270ccba0920f3bfe64d209